internal/pkg/plugin/manager.go

```diff
@@ -35,11 +36,13 @@ func _IsDynamicLibrary(file os.FileInfo) bool {
 // Manager contains the main proxy and all connection managers, listener factories,
 // and handler factories that are available
 type Manager struct {
-	ConnectionManagers map[string]plugin_v1.ConnectionManager
-	HandlerFactories   map[string]func(plugin_v1.HandlerOptions) plugin_v1.Handler
-	ListenerFactories  map[string]func(plugin_v1.ListenerOptions) plugin_v1.Listener
-	ProviderFactories  map[string]func(plugin_v1.ProviderOptions) (plugin_v1.Provider, error)
-	Proxy              secretless.Proxy
+	reloadMutex           sync.Mutex
```

Comment:
there are a lot of things that could be reloaded - is this a `configurationReloadMutex`?